### PR TITLE
Update Git guidance

### PR DESCRIPTION
**Pull request by ChrisBAshton**
Encapsulates:

- Updated `git push -f` guidance (https://github.com/alphagov/styleguides/pull/121)
- Explanation of importance of good commit messages (https://github.com/alphagov/styleguides/pull/117)
- Explains why we line-break content longer than about 80 characters.

This last point is something I've picked up on through osmosis but if anyone has any more tangible reasons, or software that automates the line breaking for you, I'm happy to amend the line-break instructions 🙌

### CODE REVIEW

File: source/standards/git.html.md.erb [hunk 1]
```diff
@@ -18,6 +18,16 @@ developers on your project. This includes:
 * any future developers (including yourself) who want to see why a change was
   made
 
+  Taking the time to write a good commit message is important for a few reasons.
```
Comment (timblair) [inline source/standards/git.html.md.erb]: The double-space at the beginning of this line should be removed.

Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Well spotted!

File: source/standards/git.html.md.erb [hunk 2]
```diff
@@ -131,35 +142,37 @@ branch in the repository history. This advice may be freely ignored for smaller
...
 
-You can also change git's default behaviour of pushing all tracked branches by
-doing `git config --global push.default upstream`, but you should probably get
-into the habit of typing out your intentions in full when doing a destructive
-operation like force pushing, otherwise disaster will strike when you use
-someone else's differently-configured git, or miss a step when configuring a
-new machine.
+## File content conventions
+
+Where possible, it is a good idea to keep line length to around 80 characters.
```
Comment (galund) [inline source/standards/git.html.md.erb]: This varies by language community, I think, and is inconsistent with guidance elsewhere, for example https://gds-way.cloudapps.digital/manuals/programming-languages/python/style-guide.html

File: source/standards/git.html.md.erb [hunk 3]
```diff
@@ -131,35 +142,37 @@ branch in the repository history. This advice may be freely ignored for smaller
...
+This ensures your colleagues can easily scan the contents of your file without
+having to set any word wrapping in their editor.
+
+This applies not just to code, but to documentation too. If, for example, you
```
Comment (galund) [inline source/standards/git.html.md.erb]: I used to think this, but actually having to re-flow paragraphs to 80 characters make it much _harder_ to follow the diff. Where a single word has been added, suddenly you have entirely lines changes.

So I actually think we should remove this guidance altogether.

Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: I totally agree to be honest! Just putting into guidance what I'm seeing, as it does seem to be a convention GOV.UK developers are following, but I think it's prone to inconsistency and solves a problem that doesn't really exist.

Happy to remove the guidance, though perhaps it should be put to some sort of vote/RFC 🤔

File: source/standards/git.html.md.erb [hunk 4]
```diff
@@ -18,6 +18,16 @@ developers on your project. This includes:
 * any future developers (including yourself) who want to see why a change was
   made
 
+Taking the time to write a good commit message is important for a few reasons.
+
+It's being kinder to our future selves. Like an
+[architecture decision record][adr], capturing context around a change allows
+people to understand why a particular implementation decision was made.
+ 
+It's an advertisement to work with you. Its demonstrates that you are
```
Comment (philandstuff) [inline source/standards/git.html.md.erb]:
```suggestion
It's an advertisement to work with you. It demonstrates that you are
```

File: source/standards/git.html.md.erb [hunk 5]
```diff
@@ -18,6 +18,16 @@ developers on your project. This includes:
...
+experienced enough to care about future maintainers. It's a palpable form of
```
Comment (philandstuff) [inline source/standards/git.html.md.erb]: "experienced enough" doesn't sound right - you can have lots of experience and not care, or you can be a new junior and get it straight away.  Maybe "It demonstrates that you care about future maintainers" is enough?

File: source/standards/git.html.md.erb [hunk 6]
```diff
@@ -72,6 +81,8 @@ You should leave a blank line before the rest of the commit message, which you
 should wrap at around 72 characters: this makes it easier to view commit
 messages in a terminal.
 
+Ignoring the established community guidelines for structuring commit messages
+is mildy anti-social to other git users.
```
Comment (philandstuff) [inline source/standards/git.html.md.erb]: While I completely agree with this, I don't think it's necessary here. We have the motivation at the top of the document, the rest of the document should just be matter-of-fact statements about what is expected.

File: source/standards/git.html.md.erb [hunk 7]
```diff
@@ -18,6 +18,16 @@ developers on your project. This includes:
...
 
+Taking the time to write a good commit message is important for a few reasons.
+
+It's being kinder to our future selves. Like an
+[architecture decision record][adr], capturing context around a change allows
+people to understand why a particular implementation decision was made.
+ 
+It's an advertisement to work with you. Its demonstrates that you are
+experienced enough to care about future maintainers. It's a palpable form of
+levelling up.
```
Comment (philandstuff) [inline source/standards/git.html.md.erb]: This last sentence feels unnecessary, and uses overly complicated language. (Can we get a tech writer to review?)

File: source/standards/git.html.md.erb (hunk 3 above)
Comment (philandstuff) [reply source/standards/git.html.md.erb]: While I prefer reflowing, I mainly feel that this is not a common enough problem to need "solving" in the GDS way, so I agree we just shouldn't have this section

File: source/standards/git.html.md.erb (hunk 5 above)
Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Agreed - will edit. Thanks :)

File: source/standards/git.html.md.erb (hunk 6 above)
Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Agreed 👍

File: source/standards/git.html.md.erb (hunk 7 above)
Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Agreed, I think we can remove this paragraph tbh.

File: source/standards/git.html.md.erb (hunk 3 above)
Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Removed 👍

File: source/standards/git.html.md.erb (hunk 2 above)
Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Section removed 👍

File: source/standards/git.html.md.erb [hunk 8]
```diff
@@ -18,6 +18,12 @@ developers on your project. This includes:
 * any future developers (including yourself) who want to see why a change was
   made
 
+Taking the time to write a good commit message is important for a few reasons.
```
Comment (galund) [inline source/standards/git.html.md.erb]: This still kinda duplicates the first sentence of the section, and the following para could arguably read better as an amendment to the last bullet point in the list of reasons?

Comment (ChrisBAshton) [reply source/standards/git.html.md.erb]: Agreed!

**Pull request merged**